src/java/org/apache/cassandra/index/Index.java

```diff
@@ -807,6 +807,17 @@ Indexer indexerFor(Predicate<Index> indexSelector,
         @Nullable
         QueryPlan queryPlanFor(RowFilter rowFilter);
 
+        /**
+         * Returns a new {@link QueryPlan} for the specified {@link RowFilter} and set of {@link Index}, or {@code null}
+         * if none of the indexes in this group supports the expression in the row filter.
+         *
+         * @param rowFilter a row filter
+         * @param indexes a set of indexes
+         * @return a new query plan for the specified {@link RowFilter} and {@link Index}, {@code null} otherwise
+         */
+        @Nullable
+        QueryPlan queryPlanForIndices(RowFilter rowFilter, Set<Index> indexes);
```

Comment:
I'm still thinking about how could make this the only method and get rid of `queryPlanFor(RowFilter)`

Reply:
I haven't addressed this yet, I will come back to it soon